JS/Lesson20/gulcin/exercises.js

```diff
@@ -67,4 +67,76 @@ console.log(square);
 
 // 3. Create a Vehicle class with make, model, and year properties, and a start() method that logs a message to the console indicating that the vehicle has started. Then create a Car class that extends the Vehicle class and adds a numDoors property. Override the start() method in the Car class to log a different message to the console indicating that the car has started.
 
+class Vehicle {
+    constructor(make, model, year) {
+        this._make = make;
+        this._model = model;
+        this._year = year;
+    }
+    start() {
+        return `The vehicle has started!`;
+    }
+}
+
+class Car extends Vehicle {
+    constructor(make, model, year, numDoors) {
+        super(make, model, year)
+        this.numDoors = numDoors;
+    }
+    start() {
+        return `The car has started!`;
+    }
+}
+
+const truck = new Vehicle('BM', 'Y85', 2018);
+console.log(truck);
+console.log(truck.start());
+const volvo = new Car('Volvo', 'XC90', 2021, 4);
+console.log(volvo);
+console.log(volvo.start());
+
+
 // 4. Create a BankAccount class with balance and interestRate properties, and deposit() and withdraw() methods. Add a transactionHistory property that is an array of objects representing each transaction made on the account, with properties for type (either "deposit" or "withdrawal"), amount, and date ({type: "withdrawal", amount: amount, date: new Date().toLocaleDateString()}). Implement deposit() and withdraw() methods that add new transactions to the transactionHistory array. Implement a get transactionHistory() getter method that returns a copy of the transaction history array, and a get currentBalance() getter method that calculates and returns the current balance of the account based on the transaction history.
+
+class BankAccount {
+    constructor(balance, interestRate) {
+        this._balance = balance;
+        this._interestRate = interestRate;
+        this._transactionHistory = [];
+    }
+
+    deposit(amount) {
+        return this._transactionHistory.push({ type: 'deposit', amount: amount, date: new Date().toLocaleDateString() });
```

Comment:
It would be good to check that the `amount` passed to this `function` is a positive integer to avoid potential issues. Also, it would be better to update `_balance` here otherwise making `withdraw` method would be a lot harder.

Reply:
Done!